pkg/sources/source.go

```diff
@@ -137,7 +137,7 @@ func (sp *SourceProcessor) Start(ctx context.Context) error {
 	shuffleFuncMap := make(map[string]*shuffle.Shuffle)
 	for _, edge := range sp.VertexInstance.Vertex.Spec.ToEdges {
 		if edge.ToVertexType == dfv1.VertexTypeReduceUDF && edge.GetToVertexPartitionCount() > 1 {
-			s := shuffle.NewShuffle(sp.VertexInstance.Vertex.GetName(), edge.GetToVertexPartitionCount())
+			s := shuffle.NewShuffle(edge.To, edge.GetToVertexPartitionCount())
```

Comment:
We need to replace it so that all Vertices that write to a given Reduce Vertex use the same Partitioning Seed to ensure they all partition the same. So, the `edge.To` Vertex enables them all to use the same seed.

Reply:
+1. Might be worth adding a comment to explicitly explain the rationale behind the scene, for future reference.